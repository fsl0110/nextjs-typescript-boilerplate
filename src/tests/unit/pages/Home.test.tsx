import { render } from 'tests';
import Home from 'pages';

describe('Home', () => {
  it('renders Home component', () => {
    render(<Home />);
    /*     screen.debug(); */
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
