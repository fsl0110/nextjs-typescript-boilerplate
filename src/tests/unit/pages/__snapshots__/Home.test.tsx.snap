// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code
          class="code"
        >
          pages/index.js
        </code>
      </p>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h2>
            Documentation →
          </h2>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h2>
            Learn →
          </h2>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h2>
            Examples →
          </h2>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h2>
            Deploy →
          </h2>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <span
          class="logo"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzIiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Vercel Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </span>
      </a>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code
          class="code"
        >
          pages/index.js
        </code>
      </p>
      <div
        class="grid"
      >
        <a
          class="card"
          href="https://nextjs.org/docs"
        >
          <h2>
            Documentation →
          </h2>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="card"
          href="https://nextjs.org/learn"
        >
          <h2>
            Learn →
          </h2>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="card"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h2>
            Examples →
          </h2>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="card"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h2>
            Deploy →
          </h2>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <span
          class="logo"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNzIiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Vercel Logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </span>
      </a>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Snapshot tests page renders and matches snapshot 1`] = `
"<html>
  <head>
    <meta name=\\"viewport\\" content=\\"width=device-width\\" />
    <meta charset=\\"utf-8\\" />
    <title>404<!-- -->: <!-- -->This page could not be found</title>
    <meta name=\\"next-head-count\\" content=\\"3\\" />
    <noscript data-n-css=\\"\\"></noscript>
  </head>
  <body>
    <div id=\\"__next\\">
      <div
        style=\\"
          color: #000;
          background: #fff;
          font-family: -apple-system, BlinkMacSystemFont, Roboto, 'Segoe UI',
            'Fira Sans', Avenir, 'Helvetica Neue', 'Lucida Grande', sans-serif;
          height: 100vh;
          text-align: center;
          display: flex;
          flex-direction: column;
          align-items: center;
          justify-content: center;
        \\"
      >
        <div>
          <style>
            body {
              margin: 0;
            }
          </style>
          <h1
            style=\\"
              display: inline-block;
              border-right: 1px solid rgba(0, 0, 0, 0.3);
              margin: 0;
              margin-right: 20px;
              padding: 10px 23px 10px 0;
              font-size: 24px;
              font-weight: 500;
              vertical-align: top;
            \\"
          >
            404
          </h1>
          <div
            style=\\"
              display: inline-block;
              text-align: left;
              line-height: 49px;
              height: 49px;
              vertical-align: middle;
            \\"
          >
            <h2
              style=\\"
                font-size: 14px;
                font-weight: normal;
                line-height: inherit;
                margin: 0;
                padding: 0;
              \\"
            >
              This page could not be found<!-- -->.
            </h2>
          </div>
        </div>
      </div>
    </div>
    <script id=\\"__NEXT_DATA__\\" type=\\"application/json\\">
      {
        \\"page\\": \\"/my-page\\",
        \\"query\\": {},
        \\"buildId\\": \\"next-page-tester\\",
        \\"props\\": { \\"pageProps\\": { \\"statusCode\\": 404 } }
      }
    </script>
  </body>
</html>
"
`;

exports[`Snapshot tests page renders and matches snapshot 2`] = `
"<div id=\\"__next\\">
  <div
    style=\\"
      color: #000;
      background: #fff;
      font-family: -apple-system, BlinkMacSystemFont, Roboto, 'Segoe UI',
        'Fira Sans', Avenir, 'Helvetica Neue', 'Lucida Grande', sans-serif;
      height: 100vh;
      text-align: center;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
    \\"
  >
    <div>
      <style>
        body {
          margin: 0;
        }
      </style>
      <h1
        style=\\"
          display: inline-block;
          border-right: 1px solid rgba(0, 0, 0, 0.3);
          margin: 0;
          margin-right: 20px;
          padding: 10px 23px 10px 0;
          font-size: 24px;
          font-weight: 500;
          vertical-align: top;
        \\"
      >
        404
      </h1>
      <div
        style=\\"
          display: inline-block;
          text-align: left;
          line-height: 49px;
          height: 49px;
          vertical-align: middle;
        \\"
      >
        <h2
          style=\\"
            font-size: 14px;
            font-weight: normal;
            line-height: inherit;
            margin: 0;
            padding: 0;
          \\"
        >
          This page could not be found<!-- -->.
        </h2>
      </div>
    </div>
  </div>
</div>
"
`;
